src/ota_proxy/ota_cache.py

```diff
@@ -333,82 +340,75 @@ async def _commit_cache_callback(self, meta: CacheMeta):
-        self,
-        raw_url: str,
-        *,
-        headers: Mapping[str, str],
-    ) -> tuple[AsyncIterator[bytes], CIMultiDictProxy[str]]:
-        async def _do_request() -> AsyncIterator[bytes]:
-            async with self._session.get(
-                self._process_raw_url(raw_url),
-                proxy=self._upper_proxy,
-                headers=headers,
-            ) as response:
-                yield response.headers  # type: ignore
-                # NOTE(20230803): sometimes aiohttp will raises:
-                #                 "ClientPayloadError: Response payload is not completed" exception,
-                #                 according to some posts in related issues, add a asyncio.sleep(0)
-                #                 to make event loop switch to other task here seems mitigates the problem.
-                #                 check the following URLs for details:
-                #                 1. https://github.com/aio-libs/aiohttp/issues/4581
-                #                 2. https://docs.python.org/3/library/asyncio-task.html#sleeping
-                await asyncio.sleep(0)
-                async for data, _ in response.content.iter_chunks():
-                    if data:  # only yield non-empty data chunk
-                        yield data
+    async def _do_request(
+        self, raw_url: str, headers: Mapping[str, str]
+    ) -> AsyncGenerator[bytes | CIMultiDictProxy[str]]:
+        async with self._session.get(
+            process_raw_url(raw_url, self._enable_https),
+            proxy=self._upper_proxy,
+            headers=headers,
+        ) as response:
+            yield response.headers
+            # NOTE(20230803): sometimes aiohttp will raises:
+            #                 "ClientPayloadError: Response payload is not completed" exception,
+            #                 according to some posts in related issues, add a asyncio.sleep(0)
+            #                 to make event loop switch to other task here seems mitigates the problem.
+            #                 check the following URLs for details:
+            #                 1. https://github.com/aio-libs/aiohttp/issues/4581
+            #                 2. https://docs.python.org/3/library/asyncio-task.html#sleeping
+            # NOTE(20250616): the original issue is closed as fixed since 3.10.
+            # await asyncio.sleep(0)
+            data_chunks = []
+            read_size = 0
+
+            # NOTE: also see https://docs.aiohttp.org/en/stable/streams.html#aiohttp.StreamReader.iter_chunks
+            async for data, end_of_http_chunk in response.content.iter_chunks():
+                if end_of_http_chunk:
+                    continue
+
+                data_chunks.append(data)
+                read_size += len(data)
+                if read_size > self.remote_read_buffer_size:
+                    yield b"".join(data_chunks)
+                    data_chunks = []
+                    read_size = 0
```

Comment:
Buffer remote downloading into 256KiB chunk to reduce method calls during request handling.